node/network/statement-distribution/src/vstaging/requests.rs

```diff
@@ -343,15 +391,34 @@ impl RequestManager {
 
 		res
 	}
+}
+
+/// A manager for pending responses.
+pub struct ResponseManager {
+	pending_responses: FuturesUnordered<BoxFuture<'static, TaggedResponse>>,
+}
+
+impl ResponseManager {
+	pub fn new() -> Self {
+		Self { pending_responses: FuturesUnordered::new() }
+	}
 
 	/// Await the next incoming response to a sent request, or immediately
 	/// return `None` if there are no pending responses.
-	pub async fn await_incoming(&mut self) -> Option<UnhandledResponse> {
+	pub async fn incoming(&mut self) -> Option<UnhandledResponse> {
```

Comment:
Changed this because `await_incoming().await` felt redundant IMO. It would be like `await_next().await` etc.